wire-library/wire-protoc/src/main/java/com/squareup/wire/protocwire/WireGenerator.kt

```diff
@@ -148,23 +152,46 @@ private fun parseFileDescriptor(fileDescriptor: DescriptorProtos.FileDescriptorP
   )
 }
 
-private fun parseMessage(path: List<Int>, helper: SourceCodeHelper, message: DescriptorProtos.DescriptorProto, descs: Plugin.DescriptorSource): MessageElement {
-  val nestedTypes = mutableListOf<TypeElement>()
+private fun parseEnum(path: List<Int>, helper: SourceCodeHelper, enum: DescriptorProtos.EnumDescriptorProto, descs: DescriptorSource): EnumElement {
+  val info = helper.getLocation(path)
+  val constants = mutableListOf<EnumConstantElement>()
+  for (enumValueDescriptorProto in enum.valueList) {
+    constants.add(EnumConstantElement(
+      location = info.loc,
+      name = enumValueDescriptorProto.name,
+      tag = enumValueDescriptorProto.number,
+      documentation = "",
```

Comment:
this comes from the source code info for the enum value

Reply:
Oh right!